oktapam/resource_resource_group_project_test.go

```diff
@@ -0,0 +1,210 @@
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccResourceGroupProjectCheckExists(resourceName, updatedResourceGroupProject),
+					resource.TestCheckResourceAttr(
+						resourceName, attributes.Name, updatedResourceGroupProjectName,
+					),
+					resource.TestCheckResourceAttr(
+						resourceName, attributes.NextUnixUID, "61200",
+					),
+					resource.TestCheckResourceAttr(
+						resourceName, attributes.NextUnixGID, "63400",
+					),
+					resource.TestCheckResourceAttr(
+						resourceName, attributes.SSHCertificateType, "CERT_TYPE_RSA_01",
+					),
+					resource.TestCheckResourceAttr(
+						resourceName, attributes.AccountDiscovery, "true",
+					),
+					resource.TestCheckResourceAttr(
+						resourceName, attributes.GatewaySelector, "env=test",
+					),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: testAccResourceGroupProjectImportStateId(resourceName),
+			},
+		},
+	})
+}
+
+func testAccResourceGroupProjectCheckExists(rn string, expectedResourceGroupProject *client.ResourceGroupProject) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[rn]
+		if !ok {
+			return fmt.Errorf("resource not found: %s", rn)
+		}
+
+		resourceGroupID := rs.Primary.Attributes[attributes.ResourceGroup]
+		projectID := rs.Primary.Attributes[attributes.ID]
+		pamClient := testAccProvider.Meta().(client.OktaPAMClient)
+		resourceGroupProject, err := pamClient.GetResourceGroupProject(context.Background(), resourceGroupID, projectID, false)
+		if err != nil {
+			return fmt.Errorf("error getting resource group project: %w", err)
+		} else if resourceGroupProject == nil {
+			return fmt.Errorf("resource group project does not exist")
+		}
+
+		err = insertComputedValuesForResourceGroupProject(expectedResourceGroupProject, resourceGroupProject)
+		if err != nil {
+			return err
+		}
+		comparison := pretty.Compare(expectedResourceGroupProject, resourceGroupProject)
+		if comparison != "" {
+			return fmt.Errorf("expected resource group project does not match returned resource group project.\n%s", comparison)
+		}
+		return nil
+	}
+}
+
+func insertComputedValuesForResourceGroupProject(expectedResourceGroupProject, actualResourceGroupProject *client.ResourceGroupProject) error {
+	actualResourceGroupProject.ID = expectedResourceGroupProject.ID
+	actualResourceGroupProject.ResourceGroupID = expectedResourceGroupProject.ResourceGroupID
+	actualResourceGroupProject.Team = expectedResourceGroupProject.Team
+
+	return nil
+}
+
+func testAccResourceGroupProjectCheckDestroy(resourceGroupName string, projectNames ...string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := testAccProvider.Meta().(client.OktaPAMClient)
+		resourceGroups, err := client.ListResourceGroups(context.Background())
```

Comment:
Not sure why we are checking for Resource Group instead of Resource Group Project ?

Reply:
So each of these tests create a new resource group and should be removed.  Since the resource group deletion will cascade the deletion of a project, we can just check if the resource group is deleted.  

I ended up removing each of the resource-specific methods and just use the method for checking if a resource group is destroyed.